### PR TITLE
Unpin CUDA docker image for GPU CI

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
- Follows up #12318 and reverts #12368
- Unnpins CUDA docker image for GPU CI
  - ~Updates PyTorch version from `1.8.1` to `1.11.*` in GPU CI~

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @tchaton @rohitgr7 @akihironitta @carmocca

### CODE REVIEW

File: .azure-pipelines/gpu-benchmark.yml
Comment (akihironitta) [inline .azure-pipelines/gpu-benchmark.yml:31]:
```suggestion
```

File: .azure-pipelines/gpu-tests.yml
Comment (akihironitta) [inline .azure-pipelines/gpu-tests.yml:32]:
```suggestion
```

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
`test_pruning_callback_ddp_spawn` is failing
`test_quantization[True-True-histogram]` crashes with 344 Segmentation fault

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by Borda** [comment]
@akihironitta have you progressed? :rabbit:

**Comment by akihironitta** [comment]
> UPDATE: Seems like the problem is with quantization...

@Borda Yes, as well as the following test failing due to segfault.

```
tests/callbacks/test_pruning.py::test_pruning_callback_ddp_spawn FAILED  [ 13%]
```

<details><summary>the failure output from pytest</summary>

https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=64058&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29&t=8b07f9df-ad34-5ead-2e0d-a2d35ff7ad3a&l=431

```
tests/callbacks/test_pruning.py::test_pruning_callback_ddp_spawn FAILED  [ 13%]

=================================== FAILURES ===================================
_______________________ test_pruning_callback_ddp_spawn ________________________

tmpdir = local('/tmp/pytest-of-AzDevOps_azpcontainer/pytest-0/test_pruning_callback_ddp_spaw0')

    @RunIf(min_gpus=2, skip_windows=True)
    def test_pruning_callback_ddp_spawn(tmpdir):
        train_with_pruning_callback(
>           tmpdir, use_global_unstructured=True, strategy="ddp_spawn", accelerator="gpu", devices=2
        )

tests/callbacks/test_pruning.py:180: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/callbacks/test_pruning.py:120: in train_with_pruning_callback
    trainer.fit(model)
pytorch_lightning/trainer/trainer.py:774: in fit
    self._fit_impl, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path
pytorch_lightning/trainer/trainer.py:724: in _call_and_handle_interrupt
    return self.strategy.launcher.launch(trainer_fn, *args, trainer=self, **kwargs)
pytorch_lightning/strategies/launchers/spawn.py:82: in launch
    start_method=self._start_method,
/usr/local/lib/python3.7/dist-packages/torch/multiprocessing/spawn.py:230: in spawn
    return start_processes(fn, args, nprocs, join, daemon, start_method='spawn')
/usr/local/lib/python3.7/dist-packages/torch/multiprocessing/spawn.py:188: in start_processes
    while not context.join():
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <torch.multiprocessing.spawn.ProcessContext object at 0x7f95e041ed90>
timeout = None

    def join(self, timeout=None):
        r"""
        Tries to join one or more processes in this spawn context.
        If one of them exited with a non-zero exit status, this function
        kills the remaining processes and raises an exception with the cause
        of the first process exiting.
    
        Returns ``True`` if all processes have been joined successfully,
        ``False`` if there are more processes that need to be joined.
    
        Args:
            timeout (float): Wait this long before giving up on waiting.
        """
        # Ensure this function can be called even when we're done.
        if len(self.sentinels) == 0:
            return True
    
        # Wait for any process to fail or all of them to succeed.
        ready = multiprocessing.connection.wait(
            self.sentinels.keys(),
            timeout=timeout,
        )
    
        error_index = None
        for sentinel in ready:
            index = self.sentinels.pop(sentinel)
            process = self.processes[index]
            process.join()
            if process.exitcode != 0:
                error_index = index
                break
    
        # Return if there was no error.
        if error_index is None:
            # Return whether or not all processes have been joined.
            return len(self.sentinels) == 0
    
        # Assume failure. Terminate processes that are still alive.
        for process in self.processes:
            if process.is_alive():
                process.terminate()
            process.join()
    
        # There won't be an error on the queue if the process crashed.
        failed_process = self.processes[error_index]
        if self.error_queues[error_index].empty():
            exitcode = self.processes[error_index].exitcode
            if exitcode < 0:
                name = signal.Signals(-exitcode).name
                raise ProcessExitedException(
                    "process %d terminated with signal %s" %
                    (error_index, name),
                    error_index=error_index,
                    error_pid=failed_process.pid,
                    exit_code=exitcode,
>                   signal_name=name
                )
E               torch.multiprocessing.spawn.ProcessExitedException: process 1 terminated with signal SIGSEGV

/usr/local/lib/python3.7/dist-packages/torch/multiprocessing/spawn.py:136: ProcessExitedException
----------------------------- Captured stderr call -----------------------------
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
WARNING:root:Bagua cannot detect bundled NCCL library, Bagua will try to use system NCCL instead. If you encounter any error, please run `import bagua_core; bagua_core.install_deps()` or the `bagua_install_deps.py` script to install bundled libraries.
WARNING:root:Bagua cannot detect bundled NCCL library, Bagua will try to use system NCCL instead. If you encounter any error, please run `import bagua_core; bagua_core.install_deps()` or the `bagua_install_deps.py` script to install bundled libraries.
Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/2
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/2
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 2 processes
----------------------------------------------------------------------------------------------------

LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1]
[W reducer.cpp:1050] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters, consider turning this flag off. Note that this warning may be a false positive your model has flow control causing later iterations to have unused parameters. (function operator())
[W reducer.cpp:1050] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters, consider turning this flag off. Note that this warning may be a false positive your model has flow control causing later iterations to have unused parameters. (function operator())
/__w/1/s/pytorch_lightning/trainer/connectors/data_connector.py:245: UserWarning: strategy=ddp_spawn and num_workers=0 may result in data loading bottlenecks. Consider setting num_workers>0 and persistent_workers=True
  "strategy=ddp_spawn and num_workers=0 may result in data loading bottlenecks."
/__w/1/s/pytorch_lightning/trainer/connectors/logger_connector/result.py:231: UserWarning: You called `self.log('test', ...)` in your `training_step` but the value needs to be floating point. Converting it to torch.float32.
  f"You called `self.log({self.meta.name!r}, ...)` in your `{self.meta.fx}` but the value needs to"
Applied `L1Unstructured`. Pruned: 0/3268 (0.00%) -> 644/3268 (19.71%)
Applied `L1Unstructured`. Pruned: 644/3268 (19.71%) -> 1095/3268 (33.51%)
Applied `L1Unstructured`. Pruned: 1095/3268 (33.51%) -> 1410/3268 (43.15%)
Applied `L1Unstructured`. Pruned: 1410/3268 (43.15%) -> 1631/3268 (49.91%)
Applied `L1Unstructured`. Pruned: 1631/3268 (49.91%) -> 1785/3268 (54.62%)
Applied `L1Unstructured`. Pruned: 1785/3268 (54.62%) -> 1893/3268 (57.93%)
Applied `L1Unstructured`. Pruned: 1893/3268 (57.93%) -> 1969/3268 (60.25%)
Applied `L1Unstructured`. Pruned: 1969/3268 (60.25%) -> 2022/3268 (61.87%)
Applied `L1Unstructured`. Pruned: 2022/3268 (61.87%) -> 2059/3268 (63.00%)
Applied `L1Unstructured`. Pruned: 2059/3268 (63.00%) -> 2085/3268 (63.80%)
free(): invalid next size (fast)
---------------- generated xml file: /__w/1/a/test-results.xml -----------------
```

</details>

---

I tried running this test in our cluster with the docker image but wasn't able to reproduce the behaviour so far. Then, I was trying to install gdb directly in CI this morning. I'm going to try again to install it to see its stacktrace now.

**Comment by akihironitta** [comment]
Blocked by the current unavailability of GPU CI.

**Comment by Borda** [comment]
@akihironitta about the Bagua, lets fist merge #12677 with complete fix for this special issue

**Comment by akihironitta** [comment]
I finally managed to install `gdb` in the container in azure CI following this comment (https://github.com/Microsoft/azure-pipelines-agent/issues/2043#issuecomment-687983301) and got the following stacktrace:

<details><summary>gdb output</summary>

```
Thread 16 "python" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7f41dc28d700 (LWP 1629)]
0x00007f427b42f48f in at::cuda::getCurrentCUDABlasHandle() () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cuda_cu.so

...

Thread 16 (Thread 0x7f41dc28d700 (LWP 1629)):
#0  0x00007f427b42f48f in at::cuda::getCurrentCUDABlasHandle() () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cuda_cu.so
#1  0x00007f427b41d7dd in void at::cuda::blas::gemm<float>(char, char, long, long, long, float, float const*, long, float const*, long, float, float*, long) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cuda_cu.so
#2  0x00007f427c6398f0 in at::native::(anonymous namespace)::addmm_out_cuda_impl(at::Tensor&, at::Tensor const&, at::Tensor const&, at::Tensor const&, c10::Scalar, c10::Scalar) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cuda_cu.so
#3  0x00007f427c63b092 in at::native::mm_cuda(at::Tensor const&, at::Tensor const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cuda_cu.so
#4  0x00007f427b474848 in at::(anonymous namespace)::(anonymous namespace)::wrapper_mm(at::Tensor const&, at::Tensor const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cuda_cu.so
#5  0x00007f427b47488f in c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (at::Tensor const&, at::Tensor const&), &at::(anonymous namespace)::(anonymous namespace)::wrapper_mm>, at::Tensor, c10::guts::typelist::typelist<at::Tensor const&, at::Tensor const&> >, at::Tensor (at::Tensor const&, at::Tensor const&)>::call(c10::OperatorKernel*, at::Tensor const&, at::Tensor const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cuda_cu.so
#6  0x00007f42698f0596 in at::Tensor c10::Dispatcher::call<at::Tensor, at::Tensor const&, at::Tensor const&>(c10::TypedOperatorHandle<at::Tensor (at::Tensor const&, at::Tensor const&)> const&, at::Tensor const&, at::Tensor const&) const [clone .constprop.4808] () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#7  0x00007f42698f5bbf in at::mm(at::Tensor const&, at::Tensor const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#8  0x00007f426b1d991c in torch::autograd::VariableType::(anonymous namespace)::mm(at::Tensor const&, at::Tensor const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#9  0x00007f426b1d9e8f in c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (at::Tensor const&, at::Tensor const&), &torch::autograd::VariableType::(anonymous namespace)::mm>, at::Tensor, c10::guts::typelist::typelist<at::Tensor const&, at::Tensor const&> >, at::Tensor (at::Tensor const&, at::Tensor const&)>::call(c10::OperatorKernel*, at::Tensor const&, at::Tensor const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#10 0x00007f4269d427d6 in at::Tensor c10::Dispatcher::call<at::Tensor, at::Tensor const&, at::Tensor const&>(c10::TypedOperatorHandle<at::Tensor (at::Tensor const&, at::Tensor const&)> const&, at::Tensor const&, at::Tensor const&) const [clone .constprop.1487] () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#11 0x00007f4269d469ff in at::Tensor::mm(at::Tensor const&) const () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#12 0x00007f426bc18911 in torch::autograd::generated::details::mm_mat1_backward(at::Tensor const&, at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::Scalar const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#13 0x00007f426afee093 in torch::autograd::generated::AddmmBackward::apply(std::vector<at::Tensor, std::allocator<at::Tensor> >&&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#14 0x00007f426b6ab771 in torch::autograd::Node::operator()(std::vector<at::Tensor, std::allocator<at::Tensor> >&&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#15 0x00007f426b6a757b in torch::autograd::Engine::evaluate_function(std::shared_ptr<torch::autograd::GraphTask>&, torch::autograd::Node*, torch::autograd::InputBuffer&, std::shared_ptr<torch::autograd::ReadyQueue> const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#16 0x00007f426b6a819f in torch::autograd::Engine::thread_main(std::shared_ptr<torch::autograd::GraphTask> const&) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#17 0x00007f426b69f979 in torch::autograd::Engine::thread_init(int, std::shared_ptr<torch::autograd::ReadyQueue> const&, bool) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_cpu.so
#18 0x00007f42dbe29163 in torch::autograd::python::PythonEngine::thread_init(int, std::shared_ptr<torch::autograd::ReadyQueue> const&, bool) () from /usr/local/lib/python3.7/dist-packages/torch/lib/libtorch_python.so
#19 0x00007f42dce69de4 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#20 0x00007f42df681609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#21 0x00007f42df400163 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

</details>

which leads to quite a few existing issues:
- https://github.com/pytorch/pytorch/issues/56747
- https://github.com/pytorch/pytorch/issues/55669
- https://github.com/open-mmlab/mmclassification/issues/603
- https://github.com/pytorch/pytorch/issues/58135

**Comment by akihironitta** [comment]
We're now using pytorch 1.8.1 and cuda11.1 in the updated container, and there're a few options to fix this:
- to downgrade cuda to 10.2 (a workaround suggested in the issues)
- to upgrade pytorch 1.9 (or 1.10 (or even 1.11))
  - FYI, kaggle and colab currently have PyTorch 1.9 and 1.10 respectively by default.

I'm in favour of upgrading pytorch version as I guess no one really uses 1.8.1 (it's not even 1.8.2 lts)... @Borda Does this sound good? If so, shall we update it to 1.10?

**Comment by Borda** [comment]
> I'm in favour of upgrading pytorch version as I guess no one really uses 1.8.1 (it's not even 1.8.2 lts)... @Borda Does this sound good? If so, shall we update it to 1.10?

I am fine with both, `1.8.2` or `1.10.1`... for sure no reason to use `1.8.1`

File: .azure-pipelines/gpu-benchmark.yml
Comment (akihironitta) [inline .azure-pipelines/gpu-benchmark.yml:31]:
```suggestion
      image: "pytorchlightning/pytorch_lightning:base-cuda-py3.7-torch1.10"
```

File: .azure-pipelines/gpu-tests.yml
Comment (akihironitta) [inline .azure-pipelines/gpu-tests.yml:32]:
```suggestion
      image: "pytorchlightning/pytorch_lightning:base-cuda-py3.7-torch1.10"
```

**Review by akihironitta — COMMENTED** [review]
TODO: Once #12675 gets merged into master and the new docker image is pushed to the hub, apply these suggestions to upgrade the version to pt1.10.

**Comment by akihironitta** [comment]
Here's another issue... This time, pytest seems hanging during its teardown as seen in
https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=67470&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29 (see timestamp)

I've seen the behaviour in CPU testing before, so I'll try to address this over the weekend.

---

@Borda One thing which makes debugging difficult/slow is that GPU CI job sometimes queues for very long time (a few hours+) until being launched. Is this something we could address in the future? e.g. change the instance type, have dedicated instances instead of spot instances (not sure how much it'll cost though)

**Comment by akihironitta** [comment]
Just for the note for myself, the cause of the hang should be the same as #11365 where the following test case hang in pytest's teardown:

```shell
$ python -m pytest tests/test_boring.py
```

```python
# ./tests/test_boring.py
import torch.distributed as dist
from pytorch_lightning import Trainer

def test_foo(single_process_pg):
    import torch.distributed._sharded_tensor

    spec = dist._sharding_spec.ChunkShardingSpec(dim=0, placements=["rank:0/cpu"])

    # if this is not called, it hangs during the test
    x = dist._sharded_tensor.empty(spec, 10, 20)
    # if called, it hangs on pytest teardown

    trainer = Trainer(strategy="horovod")
```

File: .azure-pipelines/gpu-tests.yml
Comment (Borda) [inline .azure-pipelines/gpu-tests.yml:56]:
```suggestion
        pip install https://github.com/microsoft/DeepSpeed/archive/refs/tags/v0.6.2.zip
```

File: .azure-pipelines/gpu-tests.yml
Comment (carmocca) [inline .azure-pipelines/gpu-tests.yml:22]: I strongly suggest not increasing the timeout time considerably. If the job is taking too long because newest torch is slower, then just increase it the number of minutes it requires.

Otherwise when jobs actually hang, the GPU queue will get full

File: .azure-pipelines/gpu-tests.yml
Comment (carmocca) [inline .azure-pipelines/gpu-tests.yml:32]: As the comment says, we choose 1.8 because its PyTorch's LTS version.

IMO 1.8 should be kept until there's a new LTS (1.12?)

File: .azure-pipelines/gpu-tests.yml
Comment (carmocca) [inline .azure-pipelines/gpu-tests.yml:35]: I advocate for not increasing the shm size unnecesarily to have some visibility if usage blows up

File: .azure-pipelines/gpu-tests.yml
Comment (akihironitta) [reply .azure-pipelines/gpu-tests.yml:32]: The only reason why I'm opposed to testing only with 1.8 is that we cannot test new features that are available in newer PyTorch versions. (For example, the tests fixed in #12878 were never valid since their addition. Luckily, it didn't require any changes to the codebase though.)

I think there're a few options here:
1. run tests with 1.8 (as is)
2. run tests with 1.11
3. run tests with 1.8 **+ run only tests that require 1.{9,10,11} with 1.11**

For the above reason I mentioned, I'm in favour of 2. which is the simplest, but we can try 3. if we really want to keep testing with LTS. (To be precise, the current version we're using in GPU CI is `1.8.1`, so we need to update it to the actual LTS version `1.8.2` if we do.)

@carmocca @Borda Would love to hear your thoughts!

**Comment by akihironitta** [comment]
I'm working on all bugfixes locally and push the change here first to ensure they're valid fixes, so please keep this PR ready for review to run tests in CI in a GPU environment as well as in CPU environments.

File: .azure-pipelines/gpu-tests.yml
Comment (Borda) [reply .azure-pipelines/gpu-tests.yml:32]: I am thinking about running both 1.8 and 1.11

File: tests/standalone_tests.sh
Comment (Borda) [inline tests/standalone_tests.sh:15]: @akihironitta is this the last thing which needs to be changed/reverted?

File: .azure-pipelines/gpu-tests.yml
Comment (akihironitta) [reply .azure-pipelines/gpu-tests.yml:32]: As suggested by @Borda, we'll have both 1.8 and 1.11, and this PR will now focus on unpinning the docker image without updating it to PyTorch 1.11.

**Comment by Borda** [comment]
damm:
```
tests/callbacks/test_quantization.py::test_quantization[True-True-average] /__w/_temp/fa6e0418-e4e4-49f0-af4b-342d624a344e.sh: line 1:   395 Segmentation fault
```

**Comment by akihironitta** [comment]
The failing tests are due to https://github.com/PyTorchLightning/pytorch-lightning/pull/12373#issuecomment-1094236598 where PyTorch 1.8 has some issue of segfault with CUDA 11.x. As a quick workaround (also suggested in the linked issues), I'm downgrading CUDA version from 11.3.1 to 10.2 in another PR.

File: tests/standalone_tests.sh
Comment (akihironitta) [inline tests/standalone_tests.sh:15]:
```suggestion
set -e
```

**Comment by akihironitta** [comment]
Experiencing another issue here while building `cuda10.2-py3.7-torch1.8`.

This time, installing horovod fails (I locally confirmed the issue but we'll see it fail in our CI soon). I'm pretty sure that it's due to #3449 because:
- the error looks the same
- the configuration is the same as one at the time of #12234, and
- installing 0.24.* fails but 0.23.0 succeeds.

File: .azure-pipelines/gpu-tests.yml
Comment (akihironitta) [inline .azure-pipelines/gpu-tests.yml:16]:
```suggestion
  - "master"
```

**Comment by akihironitta** [comment]
Seeing some issues with horovod tests, and here're solutions just for the note:

---

If you see
```
open MPI error: A system-required executable either could not be found or was not executable by this user 
```
try `apt install openmpi-bin` (https://github.com/openai/baselines/issues/573)

---

If you see
```
The value of the MCA parameter "plm_rsh_agent" was set to a path
that could not be found:

  plm_rsh_agent: ssh : rsh

Please either unset the parameter, or check that the path is correct
```
try `apt install ssh` (https://github.com/open-mpi/ompi/issues/3625)

File: dockers/base-cuda/Dockerfile
Comment (akihironitta) [inline dockers/base-cuda/Dockerfile:52]: We need these packages to avoid the errors pasted in https://github.com/PyTorchLightning/pytorch-lightning/pull/12373#issuecomment-1118241218.

**Comment by akihironitta** [comment]
@Borda @carmocca Would you mind having another look...?

File: .azure-pipelines/gpu-benchmark.yml
Comment (Borda) [inline .azure-pipelines/gpu-benchmark.yml:38]: let's link this with adding tests for PT 1.11

File: dockers/base-cuda/Dockerfile
Comment (Borda) [inline dockers/base-cuda/Dockerfile:18]: maybe, lets add ubuntu version as argument

**Review by Borda — COMMENTED** [review]
cool, THANK YOU @akihironitta !

File: dockers/base-cuda/Dockerfile
Comment (akihironitta) [reply dockers/base-cuda/Dockerfile:18]: Will address it when I get back home in a few hours!

File: .azure-pipelines/gpu-benchmark.yml
Comment (akihironitta) [reply .azure-pipelines/gpu-benchmark.yml:38]: You mean I should revert it here and do it in another PR adding 1.11 testing? If so, SGTM and will address it.

Comment (Borda) [reply .azure-pipelines/gpu-benchmark.yml:38]: lest keep this as it is and create new PR by adding an additional PT version, see https://github.com/PyTorchLightning/metrics/pull/1013

Comment (Borda) [reply .azure-pipelines/gpu-benchmark.yml:38]: see: #12984

File: dockers/base-cuda/Dockerfile
Comment (akihironitta) [reply dockers/base-cuda/Dockerfile]: Sorry for the delay. Addressed!

**Review by akihironitta — APPROVED** [review]

**Comment by akihironitta** [comment]
Looks strange to me that this PR got merged without three approvals...

**Comment by Borda** [comment]
> Looks strange to me that this PR got merged without three approvals...

it had them in past, the re-request review is not the same as dismiss already got approvals...

**Pull request merged**
